### PR TITLE
Add support for Python 3.9

### PR DESCRIPTION
**Pull request by icemac**
Drop support for Python 3.5 as its sunset is in some months.

This PR should fix https://travis-ci.org/github/zopefoundation/zopetoolkit/jobs/703067107 when released and used n ZTK.

### CODE REVIEW

File: src/zope/exceptions/tests/test_exceptionformatter.py [hunk 1]
```diff
@@ -17,6 +17,10 @@
 import sys
 
 
+_version = sys.version_info
+IS_PY39_OR_GREATER = _version.major == 3 and _version.minor >= 9
```
Comment (mgedmin) [inline src/zope/exceptions/tests/test_exceptionformatter.py:21]: Why not 

```python
IS_PY39_OR_GREATER = sys.version_info >= (3, 9)
```
?

**Comment by mgedmin** [comment]
(As usual, I have some slight reservations about claiming support for Python 3.9 in setup.py classifiers before 3.9 final is actually out, as we don't _know_ what other incompatible changes might creep in.  I've been burned in the past by pushing for 3.8-specific fixes to lxml/cython that then got reverted in newer 3.8 betas, breaking those lxml/cython releases.)

File: src/zope/exceptions/tests/test_exceptionformatter.py (hunk 1 above)
Comment (icemac) [reply src/zope/exceptions/tests/test_exceptionformatter.py]: Thank you.

**Comment by icemac** [comment]
I dropped claiming Python 3.9 support: you are right, we should wait for the final version as we do not know what will happen until then.

**Comment by icemac** [comment]
Thank you.

**Pull request merged**